api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -46,7 +46,8 @@ default String name() {
    *
    * @param namespace a namespace
    * @return a list of identifiers for tables
-   * @throws NotFoundException if the namespace is not found
+   * @throws NoSuchNamespaceException if the namespace is not found
+   * @throws IllegalArgumentException if the namespace is invalid for the catalog
```

Comment:
is this true for all catalogs?

Reply:
thinking more about this, I would probably remove `@throws IllegalArgumentException` and just have the change from `NotFoundException` to `NoSuchNamespaceException`

Reply:
`GlueCatalog` can throw a `ValidationException` (if validation is enabled) on an invalid namespace, meaning that the implementation details derive from the Javadoc that is being introduced here. 
That's why I think it's probably better to not mention throwing `IllegalArgumentException` here because implementations aren't consistent with this.

Reply:
ok. Removed `IllegalArgumentException`